server/src/modules/enquiry/enquiry.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Enquiry } from '../../types/mail-type'
+import { MailService } from '../mail/mail.service'
+import { Agency } from '../../bootstrap/sequelize'
+
+export class EnquiryService {
+  private mailService: Public<MailService>
+  constructor({ mailService }: { mailService: Public<MailService> }) {
+    this.mailService = mailService
+  }
+  emailEnquiry = async ({
+    agencyId,
+    enquiry,
+  }: {
+    agencyId: Array<string>
+    enquiry: Enquiry
+  }): Promise<void> => {
+    const agencyEmail = (await Promise.all(
+      agencyId.map(
+        async (Id) =>
+          (
+            await Agency.findOne({
+              attributes: ['email'],
+              where: { Id },
+            })
+          )?.email ?? '',
```

Comment:
This is difficult to read. Could you write it out properly please? Also figure a way to avoid the cast to `Array<string>` (really `string[]` might be more concise)